flink-streaming-java/src/test/java/org/apache/flink/streaming/util/OperatorSnapshotUtil.java

```diff
@@ -79,6 +81,18 @@ public static void writeStateHandle(OperatorSubtaskState state, String path) thr
 				dos.writeInt(-1);
 			}
 
+			Collection<InputChannelStateHandle> inputChannelStateHandles  = state.getInputChannelState();
+			dos.writeInt(inputChannelStateHandles.size());
+			for (InputChannelStateHandle inputChannelStateHandle : inputChannelStateHandles) {
+				MetadataV3Serializer.INSTANCE.serializeInputChannelStateHandle(inputChannelStateHandle, dos);
+			}
+
+			Collection<ResultSubpartitionStateHandle> resultSubpartitionStateHandles  = state.getResultSubpartitionState();
+			dos.writeInt(inputChannelStateHandles.size());
+			for (ResultSubpartitionStateHandle resultSubpartitionStateHandle : resultSubpartitionStateHandles) {
+				MetadataV3Serializer.INSTANCE.serializeResultSubpartitionStateHandle(resultSubpartitionStateHandle, dos);
```

Comment:
Do we need to update the `readStateHandle` function below.

Reply:
Yes, you're right. Thanks for pointing this out.